### PR TITLE
[RTD-374] Create new hashed pan container

### PR DESCRIPTION
**Pull request by petretiandrea**
This PR adds new container for hashed pans zip files.

### List of changes

- connected pipeline dataset to new container
- connected download hashpan apim to new container
- improved security for download hashpan apim by removing useless headers
- set lines for csv file to 4M (4.000.000)

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### Apply scopes
- `src` folder, only apim and storage resource -> `-target=azurerm_storage_container.cstar_hashed_pans -target=module.rtd_payment_instrument_manager_v2`
- `rtd-app`

### CODE REVIEW

**Review by GiovanniMancini — APPROVED** [review]

**Comment by GiovanniMancini** [comment]
@petretiandrea we may need a flag on new container to avoid its provisioning in production. As far as I can see, `rtd domain` is not ready for production

**Comment by petretiandrea** [comment]
> @petretiandrea we may need a flag on new container to avoid its provisioning in production. As far as I can see, `rtd domain` is not ready for production

Done

**Comment by petretiandrea** [comment]
> > @petretiandrea we may need a flag on new container to avoid its provisioning in production. As far as I can see, `rtd domain` is not ready for production
> 
> @GiovanniMancini Done

File: src/storage.tf [hunk 1]
```diff
@@ -161,6 +161,14 @@ resource "azurerm_storage_container" "cstar_exports" {
   container_access_type = "private"
 }
 
+resource "azurerm_storage_container" "cstar_hashed_pans" {
+  name                  = "cstar-hashed-pans"
+  storage_account_name  = module.cstarblobstorage.name
+  container_access_type = "private"
+
+  count = var.enable.rtd.hashed_pans_container ? 1 : 0
```
Comment (diegolagospagopa) [inline src/storage.tf:169]: Please put the count at the start of the resource

File: src/api_rtd.tf [hunk 2]
```diff
@@ -131,6 +131,9 @@ module "rtd_payment_instrument_manager" {
 module "rtd_payment_instrument_manager_v2" {
   source = "git::https://github.com/pagopa/azurerm.git//api_management_api?ref=v1.0.16"
 
+  # cause this api relies on new container, enable it when container is enabled
+  count = length(azurerm_storage_container.cstar_hashed_pans) > 0 ? 1 : 0
+
   name                = format("%s-rtd-payment-instrument-manager-api", var.env_short)
```
Comment (diegolagospagopa) [inline src/api_rtd.tf:137]: please use the string interpolation format

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]
Please check the variables for prod, because i think some one is not present

**Review by diegolagospagopa — APPROVED** [review]

**Pull request merged**